src/data/biome.ts

```diff
@@ -0,0 +1,53 @@
+// -- start tsdoc imports --
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { type getRandomWeatherType } from "#app/data/weather";
+import { type Arena } from "#app/field/arena";
+/* eslint-enable @typescript-eslint/no-unused-vars */
+// -- end tsdoc imports --
+
+import type { BiomeId } from "#enums/biome-id";
+import type { BiomePoolTier } from "#enums/biome-pool-tier";
+import type { SpeciesId } from "#enums/species-id";
+import type { TerrainType } from "#enums/terrain-type";
+import type { TimeOfDay } from "#enums/time-of-day";
+import type { TrainerType } from "#enums/trainer-type";
+import type { WeatherType } from "#enums/weather-type";
+
+/**
+ * @todo
+ * - Consider `partial` and `Omit` or an interface for `pokemonPool` and `trainerPool`
+ * - `weatherPool` and `terrainPool` are not used anywhere, move {@linkcode getRandomWeatherType} here
+ * - Add image assets here like `biomeWithProps` (in `arena.ts`)
+ * - Consider moving `mysteryEncounterByBiome` here as well
+ * - Move bgm loop point here as well
+ * - Move {@linkcode Arena.getTrainerChance} as well
+ */
+export class Biome {
+  /** The corresponding biomeId enum */
+  public biomeType: BiomeId;
```

Comment:
Why `type` and not `id`? 
Is it overwritten anywhere? Otherwise readonly, especially for `public` fields

```suggestion
  public readonly biomeId: BiomeId;
```